macrobond_data_api/util/save_proxy_to_keyring.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+
+import traceback
+import keyring
+
+from requests import get
+
+from macrobond_data_api.web.web_client import PROXY_USERNAME, DEFAULT_PROXY_SERVICE_NAME
+
+from .save_credential_to_keyring import _remove_duplicates, _test_keyring_backend
+from .transfer_performance_test import _get_url
+
+
+def save_proxy_to_keyring(warn_before_removing: bool = True, test_proxy: bool = True) -> bool:
+    # fmt: off
+    # pylint: disable=line-too-long
+    """
+    Create or update a proxy in the system's keyring interactively via terminal.
+
+    Parameters
+    ----------
+    warn_before_removing: bool
+        Optional bool whether the method should ask before removing keys, default to `True`.
+
+    test_proxy: bool
+        Optional bool whether the method should test proxy before saving, default to `True`.
+
+    Returns
+    -------
+    `bool`
+    returns `True` if the key was successful created or update in the keyring.
+
+    Remarks
+    -------
+
+    .. note::
+    It's easy to run this method via the terminal.
+    ```console  
+    python -c "from macrobond_data_api.util import *; save_proxy_to_keyring()"
+    ```
+
+    .. caution::
+    This method can remove proxy in your keyring, but it will ask first by default about warn_before_removing is `True`.
```

Comment:
This method can delete the proxy from your keyring. However, it will typically ask for your confirmation first, as the 'warn_before_removing' parameter is set to `True` by default.